napalm_eos/eos.py

```diff
@@ -985,16 +985,20 @@ def get_mac_address_table(self):
     def get_route_to(self, destination='', protocol=''):
         routes = dict()
 
+        if protocol.lower() not in ['', 'bgp', 'connected', 'isis', 'ospf', 'rip', 'static']:
+            return 'Please specify a valid protocol!'
+
         try:
             ipv = ''
             if IPNetwork(destination).version == 6:
                 ipv = 'v6'
         except AddrFormatError:
             return 'Please specify a valid destination!'
 
-        command = 'show ip{ipv} route {destination} detail'.format(
+        command = 'show ip{ipv} route {destination} {protocol} detail'.format(
```

Comment:
Here should be specify the protocol only if requested by the user. Otherwise return all possible routes.